oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/ClaimAccessor.java

```diff
@@ -98,8 +101,24 @@ default Instant getClaimAsInstant(String claim) {
 		if (Instant.class.isAssignableFrom(claimValue.getClass())) {
 			return (Instant) claimValue;
 		}
+		if (String.class.isAssignableFrom(claimValue.getClass())) {
+			try {
+				return Instant.from(DateTimeFormatter.ISO_INSTANT.parse((String)claimValue));
+			}
+			catch(DateTimeParseException e){
+				try {
+					return ZonedDateTime.from(DateTimeFormatter.ISO_ZONED_DATE_TIME.parse((String)claimValue))
+							.toInstant();
+				}
+				catch(DateTimeParseException e2){
```

Comment:
Can you ensure the code formatting is consistent with the class for try/catch - see `getClaimAsURL`